utils/merge_and_push_translations.sh

```diff
@@ -57,17 +57,29 @@ echo "Working tree is clean, continuing"
 # Print all commands.
 set -x
 
+# Pull All translations from Transifex
+# use git timestamps because the timestamps on disk are too new
+tx pull -a --use-git-timestamps
+
+if [ -n "$(git status -s)" ]; then
+  # Stage and commit translations
+  git add po/*/*.po data/i18n/locales/*.json xdg/translations/*.json || true
+  git commit -m "Fetched translations."
+fi
+
+# -----------------------
+
 # Update source catalogs
 python3 utils/buildcat.py
 
 undo_oneliner_diffs() {
   # Undo one-liner diffs of pure timestamps with no other content
   set +x
-  for entry in $(git diff --numstat po/ | sed -En 's/^1\t1\t//p'); do
-    if [ -z "$(git diff "$entry" | grep '^[+-][^+-]' | grep -v '^[+-]"POT-Creation-Date:')" ]
-    then
+  git diff --numstat po/ | sed -En 's/^1\t1\t//p' | while IFS= read -r entry; do
+    if ! git diff "$entry" | grep '^[+-][^+-]' | grep -qv '^[+-]"POT-Creation-Date:'
```

Comment:
Probably it's only me (and I understand this is the fault of the shell syntax), but it feels icky (or at least hard to read) to depend on the return value of the last element of a long pipeline in if.

Reply:
Looking at it now I find it hard to read as well. Trying ...

Reply:
I reverted it. An idea will be in a new pr.